avocado/utils/linux_modules.py

```diff
@@ -96,7 +101,22 @@ def loaded_module_info(module_name):
     :rtype: dict
     """
     l_raw = process.system_output('/sbin/lsmod')
-    return parse_lsmod_for_module(l_raw, module_name)
+    modinfo_dic = parse_lsmod_for_module(l_raw, module_name)
+    output = process.system_output("/sbin/modinfo %s" % module_name)
+    if output:
+        param_list = []
+        for line in output.splitlines():
```

Comment:
OK so this is suppose to extend the output of `parse_lsmod_for_module` info of info obtained from `modinfo`, right? It looks good and adds a lot of new info. One thing I'd change is:

```
diff --git a/avocado/utils/linux_modules.py b/avocado/utils/linux_modules.py
index 4579eaa..3bb4494 100644
--- a/avocado/utils/linux_modules.py
+++ b/avocado/utils/linux_modules.py
@@ -106,14 +106,15 @@ def loaded_module_info(module_name):
     if output:
         param_list = []
         for line in output.splitlines():
-            if line.split()[0] == 'filename:':
+            key = line.split()[0]
+            if key == 'filename:':
                 modinfo_dic['filename'] = line.split()[-1]
-            if line.split()[0] == 'version:':
+            elif key == 'version:':
                 modinfo_dic['version'] = str(line.split()[-1])
-            if line.split()[0] == 'depends:':
+            elif key == 'depends:':
                 if len(line.split()) > 1:
                     modinfo_dic['depends'] = line.split()[1].split(',')
-            if line.split()[0] == 'parm:':
+            elif key == 'parm:':
                 param_list.append(line.split()[1].split(':')[0])
         modinfo_dic['params'] = param_list
     return modinfo_dic
```

And then I'd consider how do you want to use the params. It might be actually useful to have the current setting as well as the list of params, but this won't be easy to do if the `params` is a list. How about making it a dict instead and use `open("/sys/module/%s/parameters/%s" % (module_name, line.split()[1].split(':')[0])` to get the value?